python/pyarrow/_orc.pyx

```diff
@@ -36,7 +36,233 @@ from pyarrow.lib cimport (check_status, _Weakrefable,
+        else:
+            raise ValueError("Invalid ORC writer batch size: {0}"
+                             .format(batch_size))
+
+    # file_version
+
+    if file_version is not None:
+        if str(file_version) == "0.12":
+            deref(options).file_version = FileVersion(0, 12)
+        elif str(file_version) == "0.11":
+            deref(options).file_version = FileVersion(0, 11)
+        else:
+            raise ValueError("Unsupported ORC file version: {0}"
+                             .format(file_version))
+
+    # stripe_size
+
+    if stripe_size is not None:
+        if isinstance(stripe_size, int) and stripe_size > 0:
+            deref(options).stripe_size = stripe_size
+        else:
+            raise ValueError("Invalid ORC stripe size: {0}"
+                             .format(stripe_size))
+
+    # compression
+
+    if compression is not None:
+        if isinstance(compression, basestring):
+            deref(options).compression = compression_type_from_name(
+                compression)
+        else:
+            raise TypeError("Unsupported ORC compression type: {0}"
+                            .format(compression))
+
+    # compression_block_size
+
+    if compression_block_size is not None:
+        if (isinstance(compression_block_size, int) and
+                compression_block_size > 0):
+            deref(options).compression_block_size = compression_block_size
+        else:
+            raise ValueError("Invalid ORC compression block size: {0}"
+                             .format(compression_block_size))
+
+    # compression_strategy
+
+    if compression_strategy is not None:
+        if isinstance(compression, basestring):
+            deref(options).compression_strategy = \
+                compression_strategy_from_name(compression_strategy)
+        else:
+            raise TypeError("Unsupported ORC compression strategy: {0}"
+                            .format(compression_strategy))
+
+    # row_index_stride
+
+    if row_index_stride is not None:
+        if isinstance(row_index_stride, int) and row_index_stride > 0:
+            deref(options).row_index_stride = row_index_stride
+        else:
+            raise ValueError("Invalid ORC row index stride: {0}"
+                             .format(row_index_stride))
+
+    # padding_tolerance
+
+    if padding_tolerance is not None:
+        try:
+            padding_tolerance = float(padding_tolerance)
+            deref(options).padding_tolerance = padding_tolerance
+        except Exception:
+            raise ValueError("Invalid ORC padding tolerance: {0}"
+                             .format(padding_tolerance))
+
+    # dictionary_key_size_threshold
+
+    if dictionary_key_size_threshold is not None:
+        try:
+            dictionary_key_size_threshold = float(
+                dictionary_key_size_threshold)
+            deref(options).dictionary_key_size_threshold = \
+                dictionary_key_size_threshold
```

Comment:
Should you check the value is in [0, 1] here?

Reply:
I'm almost certain that it should be in [0, 1]. Yes I have just added it.